### PR TITLE
[improve][client] Add deprecated note to Topics interface.

### PR DESCRIPTION
**Pull request by shibd**
### Motivation

Add comments to the deprecated method in the `Topics` interface.

### Modifications

- Add comments to the deprecated method in the `Topics` interface.

### Documentation

- [x] `doc-not-needed`

### CODE REVIEW

**Review by codelipenghui — APPROVED** [review]

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java [hunk 1]
```diff
@@ -2652,6 +2709,7 @@ void setInactiveTopicPolicies(String topic
      * @param topic
      * @return
      * @throws PulsarAdminException
+     * @deprecated Use {@link TopicPolicies#getMaxUnackedMessagesOnSubscriptionAsync(String, boolean)} instead.
```
Comment (MarvinCai) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java:2712]: should be `getMaxUnackedMessagesOnSubscription(String, boolean)`?

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java [hunk 2]
```diff
@@ -3223,6 +3335,7 @@ void setInactiveTopicPolicies(String topic
      * Get the compactionThreshold for a topic asynchronously. The maximum number of bytes
      * can have before compaction is triggered. 0 disables.
      * @param topic Topic name
+     * @deprecated Use {@link TopicPolicies#getCompactionThreshold(String, boolean)} instead.
```
Comment (MarvinCai) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java:3338]: getCompactionThresholdAsync(String, boolean)?

Comment (MarvinCai) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java:3338]:
```suggestion
     * @deprecated Use {@link TopicPolicies#getCompactionThresholdAsync(String, boolean)} instead.
```

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java [hunk 3]
```diff
@@ -3697,6 +3851,7 @@ void setInactiveTopicPolicies(String topic
      * @param topic Topic name
      * @param maxConsumers Max number of consumer
      * @throws PulsarAdminException Unexpected error
+     * @deprecated Use {@link TopicPolicies#setMaxConsumers(String, int)} instead.
```
Comment (MarvinCai) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java:3854]:
```suggestion
     * @deprecated Use {@link TopicPolicies#setMaxConsumersAsync(String, int)} instead.
```

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java (hunk 1 above)
Comment (MarvinCai) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java:2712]:
```suggestion
     * @deprecated Use {@link TopicPolicies#getMaxUnackedMessagesOnSubscription(String, boolean)} instead.
```

**Comment by shibd** [comment]
@MarvinCai Thanks for your review.

**Comment by dave2wave** [comment]
I what PIP was this Topics Interface deprecated?

**Comment by merlimat** [comment]
> In what PIP was this Topics Interface deprecated?

The methods were already deprecated in the past. This is only updating the Javadoc to give info on what method to use instead.

**Comment by shibd** [comment]
> In what PIP was this Topics Interface deprecated?

Here https://github.com/apache/pulsar/pull/12034.

**Pull request merged**